backend/connector_processor/exceptions.py

```diff
@@ -49,3 +49,20 @@ def __init__(self, core_err: ConnectorError) -> None:
         super().__init__(detail=core_err.message, core_err=core_err)
         self.default_detail = core_err.message
         self.status_code = 400
+
+
+class TestConnectionException(Exception):
+    DEFAULT_MESSAGE = "Test connection failed! "
+
+    def __init__(self, message: str = DEFAULT_MESSAGE):
+        super().__init__(message)
+        self.message = message
+
+    def __str__(self) -> str:
+        return self.message
```

Comment:
@kirtimanmishrazipstack please refer other exceptions and inherit from DRF's `APIException` in the django backend

Reply:
this is handled with DRF APIexception